### PR TITLE
Do not normalize volume paths on Windows by default

### PR DESCRIPTION
**Pull request by shin-**
Add environment variable `COMPOSE_CONVERT_WINDOWS_PATHS` to enable normalization if needed (Toolbox / Machine users)

This is a proposed fix for #3923.

- It might break some user projects for people using Toolbox / Machine, but the old behavior can be enabled using the newly introduced environment variable.
- It would not break projects for "Docker for Windows" users, as the proxy for D4W does that path conversion already.
- It would fix the issue for Docker on Windows Server users, which can not be worked around currently.

### CODE REVIEW

**Comment by friism** [comment]
I smoke tested this, appears to work as advertised

**Comment by shin-** [comment]
Test failure seems to be unrelated.

**Comment by shin-** [comment]
@dnephin @aanand 
- Do we have a better name for that environment variable?
- Does the code here look sane?

Nathan said on Slack he would be open to adding that variable to the output of `docker-machine env`, which would avoid breakage for people upgrading both products at least. We probably still need a docs callout and a clear note in the changelog.
I'll try and submit a PR for the machine change, but if we can settle on a name first it'd definitely help.

**Comment by dnephin** [comment]
Maybe `COMPOSE_CONVERT_WINDOWS_PATHS` ? instead of WIN32 ?

**Comment by shin-** [comment]
`s/WIN32/WINDOWS/` done

File: tests/unit/service_test.py [hunk 1]
```diff
@@ -786,11 +786,11 @@ def setUp(self):
         self.mock_client = mock.create_autospec(docker.Client)
 
     def test_build_volume_binding(self):
-        binding = build_volume_binding(VolumeSpec.parse('/outside:/inside'))
+        binding = build_volume_binding(VolumeSpec.parse('/outside:/inside', True))
```
Comment (aanand) [inline tests/unit/service_test.py]: Does the `normalize` argument need to be included here (and in the other calls in the file)? It seems irrelevant.

File: compose/config/types.py [hunk 2]
```diff
@@ -14,6 +15,8 @@
 from compose.const import IS_WINDOWS_PLATFORM
 from compose.utils import splitdrive
 
+win32_root_path_pattern = re.compile(r'[A-Za-z]\:\\.*')
```
Comment (aanand) [inline compose/config/types.py]: This should perhaps start with `^`, although it arguably doesn't matter (a path with `C:\` somewhere in the middle is clearly invalid anyway).

**Comment by aanand** [comment]
I think this is the right thing to do. Hopefully the number of people experiencing breakage will be small.
